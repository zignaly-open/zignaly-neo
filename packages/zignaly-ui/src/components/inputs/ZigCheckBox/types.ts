import { CheckboxProps } from "@mui/material/Checkbox";
import { ReactNode } from "react";

export type ZigCheckBoxProps = CheckboxProps & {
  variant?: "contained" | "outlined";
  label?: ReactNode;
};
