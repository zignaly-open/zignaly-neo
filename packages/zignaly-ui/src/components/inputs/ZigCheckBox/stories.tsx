import { Meta, StoryObj } from "@storybook/react";
import ZigCheckBox from "./index";

const meta = {
  title: "Inputs/ZigCheckBox",
  component: ZigCheckBox,
} as Meta;
export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
