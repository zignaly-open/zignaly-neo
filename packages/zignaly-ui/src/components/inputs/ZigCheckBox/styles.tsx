import { styled, Box } from "@mui/material";

import { ReactComponent as CheckmarkIcon } from "assets/icons/checkmark-active-icon.svg";

export const Icon = styled(CheckmarkIcon)`
  width: 10px;
  height: 10px;
  position: relative;
  ${({ theme }) => `fill: ${theme.palette.highlighted};`}
`;

export const Wrapper = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: center;
  background: ${({ theme }) => theme.palette.neutral800};
  border: 1px dotted ${({ theme }) => theme.palette.neutral400};
  border-radius: 2.5px;
  position: relative;
  flex: 0 0 18px;
  width: 18px;
  height: 18px;
  overflow: hidden;
`;
