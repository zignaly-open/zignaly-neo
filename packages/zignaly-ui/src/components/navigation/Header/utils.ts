// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
