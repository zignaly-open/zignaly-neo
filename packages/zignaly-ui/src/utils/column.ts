import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
