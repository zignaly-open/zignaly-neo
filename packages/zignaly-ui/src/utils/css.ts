import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
