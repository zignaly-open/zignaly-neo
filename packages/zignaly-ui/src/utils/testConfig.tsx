import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
