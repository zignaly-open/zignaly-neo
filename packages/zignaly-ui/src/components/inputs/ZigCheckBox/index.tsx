import React from "react";

import { Wrapper, Icon } from "./styles";
import Checkbox from "@mui/material/Checkbox";
import CheckBoxIcon from "@mui/icons-material/CheckBox";
import CheckBoxOutlineBlankIcon from "@mui/icons-material/CheckBoxOutlineBlank";
import { FormControlLabel } from "@mui/material";
import { ZigTypography } from "../../../index";
import { ZigCheckBoxProps } from "./types";

function ZigCheckBox({ variant = "contained", label, ...props }: ZigCheckBoxProps) {
  return (
    <FormControlLabel
      control={
        <Checkbox
          {...props}
          checkedIcon={
            variant === "outlined" ? (
              <Wrapper>
                <Icon />
              </Wrapper>
            ) : (
              <CheckBoxIcon />
            )
          }
          icon={variant === "outlined" ? <Wrapper /> : <CheckBoxOutlineBlankIcon />}
        />
      }
      label={
        variant === "outlined"
          ? label && (
              <ZigTypography
                variant={"body2"}
                color={"neutral300"}
                id={props?.id && `${props?.id}-label`}
              >
                {label}
              </ZigTypography>
            )
          : label
      }
    />
  );
}

export default ZigCheckBox;
