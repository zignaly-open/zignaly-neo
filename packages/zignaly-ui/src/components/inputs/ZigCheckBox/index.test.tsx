import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import ZigCheckBox from ".";
import { renderWithProvidersUi } from "../../../utils/testConfig";

describe("components/inputs/ZigCheckBox", () => {
  it("renders correctly with default props", () => {
    const { container } = renderWithProvidersUi(<ZigCheckBox id={"test"} />);
    const checkboxElement = container.querySelector("#test") as Element;
    const checkboxElementStyles = getComputedStyle(checkboxElement);
    expect(checkboxElement).toBeInTheDocument();
    expect(checkboxElement).toMatchSnapshot();
    expect(checkboxElementStyles).toMatchSnapshot();
  });

  it("renders correctly with outlined variant and label", () => {
    const { container } = renderWithProvidersUi(
      <ZigCheckBox variant="outlined" label="Test Label" id={"test"} />,
    );
    const checkboxElement = container.querySelector("#test") as Element;
    const labelElement = screen.getByText("Test Label");
    expect(checkboxElement).toBeInTheDocument();
    expect(labelElement).toBeInTheDocument();
    expect(checkboxElement).toMatchSnapshot();
    expect(labelElement).toMatchSnapshot();
  });

  it("handles onChange event", async () => {
    const onChangeMock = jest.fn();
    render(<ZigCheckBox onChange={onChangeMock} />);

    const checkboxElement = screen.getByRole("checkbox");

    fireEvent.click(checkboxElement);

    await waitFor(() => {
      expect(onChangeMock).toHaveBeenCalled();
    });
  });
});
