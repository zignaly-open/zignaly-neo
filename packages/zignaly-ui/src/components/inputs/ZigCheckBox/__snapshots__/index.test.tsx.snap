// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/inputs/ZigCheckBox renders correctly with default props 1`] = `
<input
  class="PrivateSwitchBase-input css-1m9pwf3"
  data-indeterminate="false"
  id="test"
  type="checkbox"
/>
`;

exports[`components/inputs/ZigCheckBox renders correctly with default props 2`] = `
CSSStyleDeclaration {
  "0": "font",
  "1": "text-rendering",
  "10": "padding",
  "11": "background-color",
  "12": "border",
  "13": "cursor",
  "14": "margin",
  "15": "box-sizing",
  "16": "position",
  "17": "opacity",
  "18": "width",
  "19": "height",
  "2": "letter-spacing",
  "20": "top",
  "21": "left",
  "22": "z-index",
  "23": "visibility",
  "3": "word-spacing",
  "4": "line-height",
  "5": "text-transform",
  "6": "text-indent",
  "7": "text-shadow",
  "8": "display",
  "9": "text-align",
  "_importants": {
    "background-color": "",
    "border": "",
    "box-sizing": "",
    "color": "",
    "cursor": "",
    "display": "",
    "font": "",
    "height": "",
    "left": "",
    "letter-spacing": "",
    "line-height": "",
    "margin": "",
    "opacity": "",
    "padding": "",
    "position": "",
    "text-align": "",
    "text-indent": "",
    "text-rendering": "",
    "text-shadow": "",
    "text-transform": "",
    "top": "",
    "visibility": undefined,
    "width": "",
    "word-spacing": "",
    "z-index": "",
  },
  "_length": 24,
  "_onChange": [Function],
  "_values": {
    "background-color": "white",
    "border": "2px inset",
    "border-bottom": "2px inset",
    "border-bottom-style": "inset",
    "border-bottom-width": "2px",
    "border-left": "2px inset",
    "border-left-style": "inset",
    "border-left-width": "2px",
    "border-right": "2px inset",
    "border-right-style": "inset",
    "border-right-width": "2px",
    "border-style": "inset",
    "border-top": "2px inset",
    "border-top-style": "inset",
    "border-top-width": "2px",
    "border-width": "2px",
    "box-sizing": "border-box",
    "cursor": "inherit",
    "display": "inline-block",
    "font": "-webkit-small-control",
    "font-family": "-webkit-small-control",
    "height": "100%",
    "left": "0px",
    "letter-spacing": "normal",
    "line-height": "normal",
    "margin": "0px",
    "margin-bottom": "0px",
    "margin-left": "0px",
    "margin-right": "0px",
    "margin-top": "0px",
    "opacity": "0",
    "padding": "0px",
    "padding-bottom": "0px",
    "padding-left": "0px",
    "padding-right": "0px",
    "padding-top": "0px",
    "position": "absolute",
    "text-align": "start",
    "text-indent": "0",
    "text-rendering": "auto",
    "text-shadow": "none",
    "text-transform": "none",
    "top": "0px",
    "visibility": "visible",
    "width": "100%",
    "word-spacing": "normal",
    "z-index": "1",
  },
}
`;

exports[`components/inputs/ZigCheckBox renders correctly with outlined variant and label 1`] = `
<input
  class="PrivateSwitchBase-input css-1m9pwf3"
  data-indeterminate="false"
  id="test"
  type="checkbox"
/>
`;

exports[`components/inputs/ZigCheckBox renders correctly with outlined variant and label 2`] = `
<span
  class="MuiTypography-root MuiTypography-body2 css-15j243s-MuiTypography-root"
  id="test-label"
>
  Test Label
</span>
`;
